### DISCUSSION TITLE
Retrieve all the different changes for a feature

### DISCUSSION THREAD

**Discussion by francbartoli** [koordinates/kart#486] [Q&A] [answered]
I've made for testing three edits on a single feature, now I'd like to get all the different changes applied for that particular feature identifier.

trying with:

`kart diff <first_commit> <last_commit> test_inizio:1 --output-format geojson --json-style pretty`

but it gives only the whole change while the log tells about the three different commits since the `first_commit`.

Is that possible with a single command?

**Comment by craigds** [comment, marked as answer]
Sorry we missed this a month ago.

It sounds like you are looking for: 

1. a way to make `log` produce a diff for each commit; i.e. an equivalent of `git log -p`.
2. a way to filter the log to only include commits that affected a particular feature, and only include those features in the diffs it produces. (see #496 )

Does that all sound right? None of that exists at present but I think we can work on it.

**Comment by rcoup** [comment]
This has been [implemented](https://github.com/koordinates/kart/pull/513) - it'll be in the v0.10.8 release when it happens. Feel free to [download a recent build](https://github.com/koordinates/kart/actions/workflows/build.yml?query=branch%3Amaster+is%3Asuccess) and try it out.

`kart log <dataset>:feature:<feature-primary-key>` or (simply) `kart log <dataset>:<feature-primary-key>`
